src/sentry/integrations/slack/service.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import annotations
+
+from logging import Logger, getLogger
+
+from sentry.integrations.repository import get_default_issue_alert_repository
+from sentry.integrations.repository.issue_alert import (
+    IssueAlertNotificationMessage,
+    IssueAlertNotificationMessageRepository,
+)
+from sentry.integrations.slack import BlockSlackMessageBuilder, SlackClient
+from sentry.models.activity import Activity
+from sentry.models.organization import OrganizationStatus
+from sentry.models.rule import Rule
+from sentry.notifications.notifications.activity import EMAIL_CLASSES_BY_TYPE
+from sentry.services.hybrid_cloud.integration import integration_service
+
+_default_logger = getLogger(__name__)
+
+
+class RuleDataError(Exception):
+    pass
+
+
+class SlackService:
```

Comment:
We have a lot of classes that look like they're the main slack handler 😅 Can you please add some documentation on what this handles.

Reply:
Yeah fair, this should be the main business logic handler going forward, I'll add a doc string